ejb/WFLY-17986.adoc

```diff
@@ -0,0 +1,234 @@
+= Support external EJB client access to EAP deployments on OpenShift over http
+:author:            Richard Achmatowicz
+:email:             rachmato@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+Existing EAP customers use the EJB client library to access EJB-based deployments (SFSBs, SLSBs, MDBs) on EAP
+on bare metal and such access needs to be provided whether the EAP deployment is on bare metal or on OpenShift.
+
+This analysis document aims to highlight the issues and concerns in providing EJB client access to EJB-based
+deployments on EAP in OpenShift.
+
+=== Client location and protocol
+Generally, an EJB client application may be internal or external to the OpenShift cluster, and may use any one of
+three transport protocols supported by the EJB client: remote, remote+http, or http (or their encrypted variants).
+
+This issue specifically focuses on the case where the EJB client application is:
+
+* external to the OpenShift cluster
+* the transport protocol is http (or its encrypted variant https)
+
```

Comment:
According to https://access.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/7.4/html/getting_started_with_jboss_eap_for_openshift_container_platform/eap-operator-for-automating-application-deployment-on-openshift_default#jakarta-enterprise-beans-remoting-on-openshift_default we have two options:

- https://access.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/7.4/html-single/configuration_guide/#remoting_remote_outbound_connection
- https://access.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/7.4/html-single/developing_jakarta_enterprise_beans_applications/#using_jakarta_enterprise_beans_client

The first one is for clients inside EAP, the second can work for both clients deployed on EAP and standalone clients (correct me if I am wrong);

Hence the questions:

- does EAP7-2062 focus on any specific kind of client? 
- in case it doesn't, is it necessary to test both standalone and EAP deployed clients or they might be considered equivalent here?

@rachmatowicz can you clarify this here?

Reply:
Hi Tomasso
The first link describes the two options available for looking up beans in a scenario where one OpenShift EAP deployment (represented by a cluster) needs to communicate with another OpenShift EAP deployment (represented by another cluster), and so both clusters are deployed on OpenShift. This is really an issue for EAP7-2063.
So I would say the type of client envisaged for EAP7-2062 was a standalone client only and not a client in an EAP deployment on a cluster external to the OpenShift cluster.

Reply:
However, that client should be able to lookup the proxy for the EJB bean in the OpenShift EAP deployment either by using JNDI/HTTP or by creating the proxy programmatically.